src/main/java/org/jabref/logic/importer/fileformat/BibtexParser.java

```diff
@@ -618,7 +618,11 @@ private String parseFieldContent(Field field) throws IOException {
                     throw new IOException("Error in line " + line + " or above: "
                             + "Empty text token.\nThis could be caused " + "by a missing comma between two fields.");
                 }
-                value.append('#').append(textToken).append('#');
+                if (field != StandardField.MONTH) {
```

Comment:
Although this change makes sense to me and we might change the logic to your version in the future, I would prefer if we don't touch parsing and writing of latex files as part of this PR. There are just too many things that can go wrong and we don't have extensive tests in place to capture them.

Thus, please keep the current behavior here (`month = jan` is stored as `#jan#` in BibEntry; moreover, `#jan#` should be written as `month = jan`).

Reply:
I think, this logic is also used to handle string constants (see https://tex.stackexchange.com/a/303489/9075), so we should be very cautious when touching this.

In short: A user should be able to enter a reference to a constant. 

* BibTex:  `Journal = IEEE_M_COM` <-> entry editor: `#IEEE_M_COM#`
* BibTex:  `Journal = {IEEE_M_COM}` <-> entry editor: `IEEE_M_COM`

This is a "quick hack" existing since the early days of JabRef. No one came up with another solution. Especially when joining constants with "real" text: BibTex: `Journal = IEEE_M_COM # {some more text}`